.cirrus.yml

```diff
@@ -108,12 +109,14 @@ task:
     su postgres <<-EOF
       meson setup \
         --buildtype=debug \
+        --prefix=${INSTALL_DIR} \
         -Dcassert=true -Dssl=openssl -Duuid=bsd -Dtcl_version=tcl86 -Ddtrace=auto \
         -DPG_TEST_EXTRA="$PG_TEST_EXTRA" \
         -Dextra_lib_dirs=/usr/local/lib -Dextra_include_dirs=/usr/local/include/ \
         build
     EOF
   build_script: su postgres -c 'ninja -C build -j${BUILD_JOBS}'
+  install_script: su postgres -c 'ninja -C build install'
```

Comment:
We can't require 'install' to have been done to run tests, they need to work without that. Why isn't the tmp_install installation sufficient?

Reply:
Because adminpack couldn't find 'postgres.h' so pseudo installed test fails. The cause seems pkg-conf has installation paths for finding required files, not 'tmp_install' path.

For example:

```
############################################################
Testing postgresql-extension-warnings

The Meson build system
Version: 0.63.3
Source dir: /home/nbyavuz/Desktop/projects/meson-postgres/build/contrib/adminpack_tmp
Build dir: /home/nbyavuz/Desktop/projects/meson-postgres/build/contrib/adminpack_tmp/build
Build type: native build
Project name: adminpack
Project version: undefined
C compiler for the host machine: ccache cc (gcc 12.2.1 "cc (SUSE Linux) 12.2.1 20220830 [revision e927d1cf141f221c5a32574bde0913307e140984]")
C linker for the host machine: cc ld.bfd 2.39.0.20220810-1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (1.8.0)
Run-time dependency postgresql-extension-warnings found: YES 16devel
Build targets in project: 1

Found ninja-1.11.1 at /usr/bin/ninja
ninja: Entering directory `build'
[1/2] ccache cc -Iadminpack.so.p -I. -I.. -I/tmp/meson-install/include -I/tmp/meson-install/include/postgresql/server -I/usr/include/libxml2 -I/usr/include/security -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O0 -g -fPIC -Wmissing-prototypes -Wpointer-arith -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -Wdeclaration-after-statement -Wno-format-truncation -Wno-stringop-truncation -pthread -fvisibility=hidden -D_GNU_SOURCE -fno-strict-aliasing -fwrapv -fexcess-precision=standard -DOPENSSL_LOAD_CONF -MD -MQ adminpack.so.p/_home_nbyavuz_Desktop_projects_meson-postgres_contrib_adminpack_adminpack.c.o -MF adminpack.so.p/_home_nbyavuz_Desktop_projects_meson-postgres_contrib_adminpack_adminpack.c.o.d -o adminpack.so.p/_home_nbyavuz_Desktop_projects_meson-postgres_contrib_adminpack_adminpack.c.o -c /home/nbyavuz/Desktop/projects/meson-postgres/contrib/adminpack/adminpack.c
FAILED: adminpack.so.p/_home_nbyavuz_Desktop_projects_meson-postgres_contrib_adminpack_adminpack.c.o 
ccache cc -Iadminpack.so.p -I. -I.. -I/tmp/meson-install/include -I/tmp/meson-install/include/postgresql/server -I/usr/include/libxml2 -I/usr/include/security -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O0 -g -fPIC -Wmissing-prototypes -Wpointer-arith -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -Wdeclaration-after-statement -Wno-format-truncation -Wno-stringop-truncation -pthread -fvisibility=hidden -D_GNU_SOURCE -fno-strict-aliasing -fwrapv -fexcess-precision=standard -DOPENSSL_LOAD_CONF -MD -MQ adminpack.so.p/_home_nbyavuz_Desktop_projects_meson-postgres_contrib_adminpack_adminpack.c.o -MF adminpack.so.p/_home_nbyavuz_Desktop_projects_meson-postgres_contrib_adminpack_adminpack.c.o.d -o adminpack.so.p/_home_nbyavuz_Desktop_projects_meson-postgres_contrib_adminpack_adminpack.c.o -c /home/nbyavuz/Desktop/projects/meson-postgres/contrib/adminpack/adminpack.c
/home/nbyavuz/Desktop/projects/meson-postgres/contrib/adminpack/adminpack.c:15:10: fatal error: postgres.h: No such file or directory
   15 | #include "postgres.h"
      |          ^~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

Reply:
Sure, but the installation in tmp_install/ will have postgres.h - isn't this just a question of referencing that?

Reply:
Yes. It is set in `postgresql-extension-warnings.pc` file:

```
prefix=/usr/local/pgsql
includedir=${prefix}/include

build_inc0=""
src_inc0=""
pkglibdir=${prefix}/lib/x86_64-linux-gnu
dir_mod=${pkglibdir}
dir_data=${prefix}/share/extension
bindir=${prefix}/bin

Name: postgresql-extension-warnings
Description: PostgreSQL Extension Support - Compiler Warnings
URL: https://www.postgresql.org/
Version: 16beta1
Requires: postgresql-extension
Cflags: -I${includedir} -Wmissing-prototypes -Wpointer-arith -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -Wdeclaration-after-statement -Wno-format-truncation -Wno-stringop-truncation
```

As you said earlier, we can set `-Dpkgconfig.relocatable=true` but [this option](https://mesonbuild.com/Builtin-options.html#pkgconfig-module) is introduced since meson 0.63.0.

Reply:
I don't know if you would prefer it, but I was trying a couple of ways and I solved that using a bit of an hacky way :( I edited prefix part in the .pc file while pkg_config_tests were running, so that it will reference where postgres files are.

Test passed on all tasks except Windows and Centos 8. I didn't debug Centos 8 yet. It couldn't find Pkg-config binary on Windows, so I disabled this test on Windows for now.

[CI Run](https://cirrus-ci.com/build/6244741092737024)
[Commit](https://github.com/nbyavuz/postgres/commit/2979b0a118e200f21cf5dd671a8b63314434c20e)
[Edit .pc part](https://github.com/nbyavuz/postgres/blob/2979b0a118e200f21cf5dd671a8b63314434c20e/src/tools/ci/test_pkg_conf#LL115C1-L133C24)